local-interchaintest/src/main.rs

```diff
@@ -0,0 +1,161 @@
+};
+
+mod utils;
+
+pub const POLYTONE_PATH: &str = "local-interchaintest/wasms/polytone";
+pub const LOGS_FILE_PATH: &str = "local-interchaintest/configs/logs.json";
+pub const LOCAL_CODE_ID_CACHE_PATH_NEUTRON: &str =
+    "local-interchaintest/code_id_cache_neutron.json";
+
+pub const ACC0_KEY: &str = "acc0";
+pub const ACC0_ADDR: &str = "neutron1hj5fveer5cjtn4wd6wstzugjfdxzl0xpznmsky";
+pub const ACC1_KEY: &str = "acc1";
+pub const ACC1_ADDR: &str = "neutron1kljf09rj77uxeu5lye7muejx6ajsu55cuw2mws";
+pub const ACC2_KEY: &str = "acc2";
+pub const ACC2_ADDR: &str = "neutron17lp3n649rxt2jadn455frcj0q6anjnds2s0ve9";
+
+pub const GAS_FLAGS: &str = "--gas=auto --gas-adjustment=3.0";
+
+fn main() -> Result<(), Box<dyn Error>> {
+    env_logger::init();
+
+    let mut test_ctx = TestContextBuilder::default()
+        .with_unwrap_raw_logs(true)
+        .with_api_url("http://localhost:42069/")
+        .with_artifacts_dir("artifacts")
+        .with_chain(ConfigChainBuilder::default_neutron().build()?)
+        .with_chain(ConfigChainBuilder::default_gaia().build()?)
+        .with_chain(ConfigChainBuilder::default_juno().build()?)
+        .with_log_file_path(LOGS_FILE_PATH)
+        .with_transfer_channels(NEUTRON_CHAIN_NAME, GAIA_CHAIN_NAME)
+        .with_transfer_channels(NEUTRON_CHAIN_NAME, JUNO_CHAIN_NAME)
+        .build()?;
+
+    let mut uploader = test_ctx.build_tx_upload_contracts();
+
+    // TODO: uncomment to deploy polytone
+    // uploader
+    //     .with_key(ACC0_KEY)
+    //     .send_with_local_cache(POLYTONE_PATH, LOCAL_CODE_ID_CACHE_PATH_NEUTRON)
+    //     .unwrap();
+
+    let current_dir = env::current_dir()?;
+
+    let orbital_core_local_path = format!("{}/artifacts/orbital_core.wasm", current_dir.display());
+
+    uploader
+        .with_chain_name(NEUTRON_CHAIN_NAME)
+        .send_single_contract(&orbital_core_local_path)?;
+
+    let orbital_core_code_id = test_ctx
+        .get_contract()
+        .contract("orbital_core")
+        .get_cw()
+        .code_id
+        .unwrap();
+
+    info!("orbital core code id: {orbital_core_code_id}");
+
+    let orbital_instantiate_msg = InstantiateMsg {
+        owner: test_ctx
+            .get_chain(NEUTRON_CHAIN_NAME)
+            .admin_addr
+            .to_string(),
+    };
+
+    // instantiate orbital-core from the ACC0_KEY (=admin in localic-utils)
+    let orbital_core = contract_instantiate(
+        test_ctx
+            .get_request_builder()
+            .get_request_builder(NEUTRON_CHAIN_NAME),
+        ACC0_KEY,
+        orbital_core_code_id,
+        &serde_json::to_string(&orbital_instantiate_msg)?,
+        "orbital_core",
+        None,
+        "",
+    )?;
+
+    info!("orbital core: {}", orbital_core.address);
+
+    admin_register_domain(
+        &test_ctx,
+        orbital_core.address.to_string(),
+        GAIA_CHAIN_NAME.to_string(),
+    )?;
+    std::thread::sleep(Duration::from_secs(5));
```

Comment:
Isn't there a way to check if whatever we are waiting for is finished? or force it to finish? 
Basically being proactive instead of sleeping

Reply:
my bad, some of these were leftovers from debugging!